src/append_only_data.rs

```diff
@@ -931,7 +960,7 @@ mod tests {
     #[test]
     fn seq_append_entries() {
         let mut data = SeqAppendOnlyData::<PubPermissions>::new(XorName([1; 32]), 10000);
-        let res = data.append(&[(b"hello".to_vec(), b"world".to_vec())], 0);
+        let res = data.append([(b"hello".to_vec(), b"world".to_vec())].to_vec(), 0);
```

Comment:
A minor point: this could be `vec![...]` rather than `[...].to_vec()`.

Also, we seem to have a few instances of
```rust
match res {
    Ok(()) => (),
    Err(x) => panic!("Unexpected error: {:?}", x),
}
```
These could all be removed if we just do e.g.
```rust
unwrap!(data.append(vec![(b"hello".to_vec(), b"world".to_vec())], 0));
```